celery/worker/loops.py

```diff
@@ -71,15 +71,7 @@ def asynloop(obj, connection, consumer, blueprint, hub, qos,
 
     try:
         while blueprint.state == RUN and obj.connection:
-            # shutdown if signal handlers told us to.
-            should_stop, should_terminate = (
-                state.should_stop, state.should_terminate,
-            )
-            # False == EX_OK, so must use is not False
-            if should_stop is not None and should_stop is not False:
-                raise WorkerShutdown(should_stop)
-            elif should_terminate is not None and should_stop is not False:
```

Comment:
`should_stop` in this line is the root cause of the issue.
It should be `should_terminate`, just like in the implementation of the `state.maybe_shutdown` method.